clients/admin-ui/src/features/config-wizard/ScanResultsForm.tsx

```diff
@@ -33,26 +36,33 @@ const ALL_COLUMNS: ColumnMetadata[] = [
 const ScanResultsForm = () => {
   const systems = useAppSelector(selectSystemsForReview);
   const dispatch = useAppDispatch();
+  const router = useRouter();
   const {
     isOpen: isWarningOpen,
     onOpen: onWarningOpen,
     onClose: onWarningClose,
   } = useDisclosure();
+  const [upsertSystems] = useUpsertSystemsMutation();
   const [selectedSystems, setSelectedSystems] = useState<System[]>(systems);
+  const features = useFeatures();
   const [selectedColumns, setSelectedColumns] =
     useState<ColumnMetadata[]>(ALL_COLUMNS);
 
+  const createSystems = async () => {
+    await upsertSystems(selectedSystems);
+  };
+
   const confirmRegisterSelectedSystems = () => {
     dispatch(chooseSystemsForReview(selectedSystems.map((s) => s.fides_key)));
-    dispatch(changeStep());
+    createSystems();
+    return features.plus ? router.push(`/datamap`) : router.push(`/system`);
   };
 
   const handleSubmit = () => {
     if (systems.length > selectedSystems.length) {
-      onWarningOpen();
-    } else {
-      confirmRegisterSelectedSystems();
+      return onWarningOpen();
     }
+    return confirmRegisterSelectedSystems();
```

Comment:
same thing here, I don't think you need the `return`. not that it hurts, but it makes me think something other than void is getting returned